q2_sample_classifier/utilities.py

```diff
@@ -78,6 +78,9 @@ def _load_data(feature_data, targets_metadata):
     # Load metadata, attempt to convert to numeric
     targets = targets_metadata.to_dataframe()
 
+    if missing_samples == 'error':
+        _validate_metadata_is_superset(targets, feature_data)
+
     # filter features and targets so samples match
     merged = feature_data.join(targets, how='inner')
```

Comment:
Can we add one test for the util `_load_data` that ensures that an inner happens when missing_samples == 'ignore'. I don't think we want to be in the business of testing pandas join behavior, but rather, adding this test in as a sanity check for the future, in case some endeavoring developer decides to change the `how` parameter to some other method, we at least will have a little warning that this behavior is changing!

Reply:
thanks @thermokarst ! I have made both changes as requested.

(also note there are a few additional changes in the most recent commit, related to merging master and adding missing sample handling to the NCV methods that were just merged to master. You can probably ignore those changes.)